### ISSUE TITLE
Flowgraph window issues

### ISSUE THREAD

**Issue by ghramsey** [MattFiler/OpenCAGE#382]
First congrats on this update I feel like it will be great once all worked out.

I dropped a video link since it further illustrates some of the issues.  
I tried to show the menu settings for context too.

The nodegraph is drawn partly off screen. There seems to be no way to scroll about the graph window. 
In the past it's been possible to select then drag the nodes from off screen into the center.  I was unable to effectively 
do that.  Moving the nodes around or at least being able to maintain focus on the nodegraph window so it can be scrolled
through would be helpful.   

If the option to change to entity is set on then you cannot  select a group to move it tries to scroll through all of the selected 
nodes and gets to the end then opens the window.  It's doing that part as partially as intended I think but it makes navigating 
the graph window harder without a way to otherwise move the node to be seen.  (This was not in the clip.)

I was also able to right click the node links and they went away.  
IDK if those are only visual or actually broke the entity links.

I tried resetting the UI in case that'd help with any of the above, but no dice.
 

https://github.com/user-attachments/assets/8ab1c16c-718b-4c1d-82e6-4ba908a7a4aa

**Comment by MattFiler** [comment]
Hey! I'll work on some documentation, but perhaps I should put a controls list directly in the app.

To navigate around the flowgraph UI:
- Hold middle mouse to move
- Right click on a link to delete it
- Click and drag between the links to link them
- Right click a node to add/remove links and duplicate/delete the node
- Right click anywhere to add a new node

**Comment by MattFiler** [comment]
Also, currently the width and pin state of the entity list and entity inspector don't persist between composites - I think adding that should hopefully solve your layout issues.

**Comment by MattFiler** [comment]
Yeah it's a slightly different flow (and admittedly confusing going between the new UI and old UI for composites that don't yet support flowgraphs), but once you get used to it it's a lot easier to see what's going on, and to script new logic. 

Definitely open to any suggestions to make it any easier to figure out!

**Comment by ghramsey** [comment]
I'll need to play around more with it to see how it works.   
MMB to scroll feels awkward and until you said to use it I didn't think to do so.  
IDK how to solve that though.

**Comment by MattFiler** [comment]
I could always add a window to change the mappings

**Comment by ghramsey** [comment]
I keep mistakenly trying to scroll the window w/ RMB.   
The RMB is by default a selection.  I wonder if that is used?
Can they be swapped in function ?  Seems scrolling might be used more often than select.
Could it be easy as adding a control or shift as modifier to switch them ?  Hold Control swaps RMB and MMB
letting go reverts to standard ? Or perhaps by num lock or caps lock.  
I might be an outlier in how I use the functions so that would cover more than just me.

Also hoping it doesn't take lots of added work to change.

**Comment by MattFiler** [comment]
I've added a new window that lists controls. I might add rebinding at a later date but I think this solves the issue for now.

I've also made it so that the widths of the entity list and entity inspector will be remembered between sessions :)
